### PR TITLE
PHP 8.2 support

### PR DESCRIPTION
**Pull request by MidnightDesign**

### CODE REVIEW

**Comment by MidnightDesign** [comment]
This also switches all GH Actions workflows that previously only ran on 8.1 to 8.2.

Waiting for the 8.2 support in [`doctrine/doctrine-module`](https://github.com/doctrine/DoctrineModule) to be tagged.

**Comment by MidnightDesign** [comment]
@driehle Mind taking a look? : )

File: composer.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     ],
     "homepage": "https://www.doctrine-project.org/",
     "require": {
-        "php": "^7.4 || ~8.0.0 || ~8.1.0",
+        "php": "^7.4 || ~8.0.0 || ~8.1.0 || ~8.2.0",
         "ext-mongodb": "*",
         "doctrine/doctrine-laminas-hydrator": "^3.1.0",
         "doctrine/doctrine-module": "^5.2.1",
```
Comment (driehle) [inline composer.json]: Could you please bump this to 5.3.0? This reduces complexity for maintainers, as less combinations of the two modules are allowed then.

Comment (MidnightDesign) [reply composer.json]: Done!

**Review by driehle — APPROVED** [review]

**Pull request merged**